src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs

```diff
@@ -78,6 +85,41 @@ internal virtual IList<string> FindPotentialSdkResolvers(string rootFolder, Elem
             return assembliesList;
         }
 
+        private DirectoryInfo[] GetSubfolders(string rootFolder, string additionalResolversFolder)
+        {
+            DirectoryInfo[] subfolders = null;
+            if (!string.IsNullOrEmpty(rootFolder) && FileUtilities.DirectoryExistsNoThrow(rootFolder))
+            {
+                subfolders = new DirectoryInfo(rootFolder).GetDirectories();
+            }
+
+            if (additionalResolversFolder != null)
+            {
+                var resolversDirInfo = new DirectoryInfo(additionalResolversFolder);
+                if (resolversDirInfo.Exists)
+                {
+                    var overrideFolders = resolversDirInfo.GetDirectories();
+                    var foldersSet = overrideFolders.ToHashSet(new DirInfoNameComparer());
```

Comment:
I wonder if we'll ever hit a scenario where someone has a ridiculous number of directories such that this impacts perf. This is more a comment than an actual suggestion.

Reply:
Note that this would also only happen with the test hook enabled.

Reply:
Also, this is now asymptotically optimal. I don't think it's likely that it _has_ to be that good, but MSBuild users continually surprise me...

Reply:
Considering what Daniel mentioned, I'm fine with this as-is.